tools/autograd/templates/python_torch_functions.cpp

```diff
@@ -485,73 +485,37 @@ static PyObject * THPVariable_frombuffer(PyObject* self_, PyObject* args, PyObje
     auto offset = r.toInt64(3);
     auto requires_grad = r.toBool(4);
 
-    auto elsize = at::elementSize(dtype);
-    size_t actual_count = 0;
-    Py_buffer view;
-
     TORCH_CHECK_VALUE(
         PyObject_CheckBuffer(buffer) != 0,
         "object does not implement Python buffer protocol.");
+    ret = wrap(torch::utils::tensor_frombuffer(
+        buffer, dtype, count, offset, requires_grad));
+  }
 
-    if (PyObject_GetBuffer(buffer, &view, PyBUF_WRITABLE) < 0) {
-      TORCH_CHECK(
-          PyObject_GetBuffer(buffer, &view, PyBUF_SIMPLE) >= 0,
-          "could not retrieve buffer from object");
-      TORCH_WARN_ONCE(
-          "The given buffer is not writable, and PyTorch does "
-          "not support non-writable tensors. This means you can write to the "
-          "underlying (supposedly non-writable) buffer using the tensor. "
-          "You may want to copy the buffer to protect its data or make it writable "
-          "before converting it to a tensor. This type of warning will be "
-          "suppressed for the rest of this program.");
-      PyErr_Clear();
-    }
-
-    Py_INCREF(view.obj);
-    THPObjectPtr obj(view.obj);
+  return ret;
 
-    auto len = view.len;
-    auto buf = view.buf;
-    PyBuffer_Release(&view);
+  Py_RETURN_NONE;
+  END_HANDLE_TH_ERRORS
+}
 
-    TORCH_CHECK_VALUE(
-        len > 0 && count != 0,
-        "both buffer length (", len, ") and count (", count, ") must not be 0");
-    TORCH_CHECK_VALUE(
-        offset >= 0 && offset < len,
-        "offset (", offset, " bytes) must be non-negative and no greater than "
-        "buffer length (", len, " bytes) minus 1");
-    TORCH_CHECK_VALUE(
-        count > 0 || (len - offset) % elsize == 0,
-        "buffer length (", len - offset, " bytes) after offset (", offset, " bytes) "
-        "must be a multiple of element size (", elsize, ")");
+static PyObject * THPVariable_asarray(PyObject* self_, PyObject* args, PyObject* kwargs)
```

Comment:
Why does `asarray` need custom parsing?

Reply:
By "custom parsing", do you mean the fact that I'm using `PythonArgParser` here? If so, it's not really related to `self`. It's just a way I found to parse the `kwargs`.